lua/neogit/integrations/diffview.lua

```diff
@@ -84,11 +95,11 @@ local function get_local_diff_view(section_name, item_name, opts)
     end,
   }
 
-  view:on_files_staged(a.void(function(_)
-    Watcher.instance():dispatch_refresh()
-    view:update_files()
-  end))
-
+  view:on_files_staged(function()
+    vim.schedule(function()
+      Watcher.instance():dispatch_refresh()
+    end)
+  end)
```

Comment:
Can you explain why this needs to be scheduled, instead of non-blocking async? Not sure I follow.